mobile/examples/whisper/azure/android/app/src/main/java/ai/onnxruntime/example/whisperAzure/MainActivity.kt

```diff
@@ -0,0 +1,188 @@
+package ai.onnxruntime.example.whisperAzure
+
+import android.Manifest
+import android.content.pm.PackageManager
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.widget.Button
+import android.widget.TextView
+import android.widget.Toast
+import androidx.core.app.ActivityCompat
+import java.util.concurrent.Executors
+import java.util.concurrent.atomic.AtomicBoolean
+
+class MainActivity : AppCompatActivity() {
+    private val usePrerecordedAudioButton: Button by lazy { findViewById(R.id.use_prerecorded_audio_button) }
+    private val recordAudioButton: Button by lazy { findViewById(R.id.record_audio_button) }
+    private val stopRecordingAudioButton: Button by lazy { findViewById(R.id.stop_recording_audio_button) }
+
+    private val resultText: TextView by lazy { findViewById(R.id.result_text) }
+    private val statusText: TextView by lazy { findViewById(R.id.status_text) }
+
+    private val speechRecognizer: SpeechRecognizer by lazy {
+        resources.openRawResource(R.raw.openai_whisper_transcriptions).use {
+            val modelBytes = it.readBytes()
+            SpeechRecognizer(modelBytes)
+        }
+    }
+
+    private val stopRecordingFlag = AtomicBoolean(false)
+
+    private val workerThreadExecutor = Executors.newSingleThreadExecutor()
+
+    private fun setSuccessfulResult(result: SpeechRecognizer.Result) {
+        runOnUiThread {
+            statusText.text = "Successful speech recognition (${result.inferenceTimeInMs} ms)"
+            resultText.text = result.text.ifEmpty { "<No speech detected.>" }
+        }
+    }
+
+    private fun setError(exception: Exception) {
+        Log.e(TAG, "Error: ${exception.localizedMessage}", exception)
+        runOnUiThread {
+            statusText.text = "Error"
+            resultText.text = exception.localizedMessage
+        }
+    }
+
+    private fun setError(result: SpeechRecognizer.Result) {
+        Log.i(TAG, "Error: ${result.text}")
```

Comment:
nit: can we also log at error level to be consistent with the other `setError`?
```suggestion
        Log.e(TAG, "Error: ${result.text}")
```